source/app/helpers/sessions_helper.rb

```diff
@@ -1,2 +1,18 @@
 module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def is_admin?
+    return false if current_user.nil?
+    current_user.user_type == 'admin'
```

Comment:
Thanks for your help Mike, awesome stuff. Definitely learned some things here. I had been wondering why my signup error messages from @user.errors had stopped displaying and hadn't guessed it could be related to the fact that I switched from rendering to redirecting in my User controller. Neat